test/k8sT/Services.go

```diff
@@ -479,29 +482,55 @@ var _ = Describe("K8sServicesTest", func() {
 			cmdIn := "cilium bpf ct list global | awk '/%s/ { sub(\".*=\",\"\", $7); print $7 }'"
 
 			endpointK8s1 := fmt.Sprintf("%s:%d", dstPodIPK8s1, dstPodPort)
-			patternInK8s1 := fmt.Sprintf("UDP IN %s:%d -> %s", srcIP, srcPort, endpointK8s1)
+			newSrcIP1 := srcIP
+			newSrcIP2 := srcIP
+			if kubeProxy {
+				// If kube-proxy is enabled, source IP address are SNATed.
+				if strings.HasSuffix(dstIP, k8s2IP) {
+					newSrcIP1, err = kubectl.GetCiliumHostIPv4(context.TODO(), helpers.K8s2)
+					Expect(err).Should(BeNil(), "Cannot retrieve remote cilium_host ipv4")
+					newSrcIP2 = hostIP
+				} else {
+					newSrcIP1 = hostIP
+					newSrcIP2, err = kubectl.GetCiliumHostIPv4(context.TODO(), helpers.K8s1)
```

Comment:
For the `fromNode == helpers.K8s2` case:
Here we are basically inverting `newSrcIP1` and `newSrcIP2` based on whether the destination is on `k8s1` or `k8s2`. I'm expecting this does not actually depend on `k8s1` vs. `k8s2`, but on whether the destination is on the same node as the source or not. So this depends on `fromNode`.

However, as far as I can see, the tests do not make use of the `fromNode` parameter, so I'm wondering if we should just remove it and consider we always send from `k8s1`.

Reply:
Ok, thanks. Yeah, you're probably right, we can try that.